// mysql2/lib/constants/errors.js
// This file was modified by Oracle on June 1, 2021.
// An entry was created for a new error reported by the MySQL server due to
// client inactivity.
// Modifications copyright (c) 2021, Oracle and/or its affiliates.

'use strict';

// originally copied from https://raw.githubusercontent.com/mysqljs/mysql/7770ee5bb13260c56a160b91fe480d9165dbeeba/lib/protocol/constants/errors.js
// (c) node-mysql authors

// updated to contain error codes as is contained in MySQL 8.0
// by adapting node-mysql: /.../generate-error-constants.js

/**
 * MySQL error constants
 * 
 * Extracted from version 8.0.33
 * 
 * !! Generated by generate-error-constants.js, do not modify by hand !!
 */


exports.EE_CANTCREATEFILE = 1
exports.EE_READ = 2
exports.EE_WRITE = 3
exports.EE_BADCLOSE = 4
exports.EE_OUTOFMEMORY = 5
exports.EE_DELETE = 6
exports.EE_LINK = 7
exports.EE_EOFERR = 9
exports.EE_CANTLOCK = 10
exports.EE_CANTUNLOCK = 11
exports.EE_DIR = 12
exports.EE_STAT = 13
exports.EE_CANT_CHSIZE = 14
exports.EE_CANT_OPEN_STREAM = 15
exports.EE_GETWD = 16
exports.EE_SETWD = 17
exports.EE_LINK_WARNING = 18
exports.EE_OPEN_WARNING = 19
exports.EE_DISK_FULL = 20
exports.EE_CANT_MKDIR = 21
exports.EE_UNKNOWN_CHARSET = 22
exports.EE_OUT_OF_FILERESOURCES = 23
exports.EE_CANT_READLINK = 24
exports.EE_CANT_SYMLINK = 25
exports.EE_REALPATH = 26
exports.EE_SYNC = 27
exports.EE_UNKNOWN_COLLATION = 28
exports.EE_FILENOTFOUND = 29
exports.EE_FILE_NOT_CLOSED = 30
exports.EE_CHANGE_OWNERSHIP = 31
exports.EE_CHANGE_PERMISSIONS = 32
exports.EE_CANT_SEEK = 33
exports.EE_CAPACITY_EXCEEDED = 34
exports.EE_DISK_FULL_WITH_RETRY_MSG = 35
exports.EE_FAILED_TO_CREATE_TIMER = 36
exports.EE_FAILED_TO_DELETE_TIMER = 37
exports.EE_FAILED_TO_CREATE_TIMER_QUEUE = 38
exports.EE_FAILED_TO_START_TIMER_NOTIFY_THREAD = 39
exports.EE_FAILED_TO_CREATE_TIMER_NOTIFY_THREAD_INTERRUPT_EVENT = 40
exports.EE_EXITING_TIMER_NOTIFY_THREAD = 41
exports.EE_WIN_LIBRARY_LOAD_FAILED = 42
exports.EE_WIN_RUN_TIME_ERROR_CHECK = 43
exports.EE_FAILED_TO_DETERMINE_LARGE_PAGE_SIZE = 44
exports.EE_FAILED_TO_KILL_ALL_THREADS = 45
exports.EE_FAILED_TO_CREATE_IO_COMPLETION_PORT = 46
exports.EE_FAILED_TO_OPEN_DEFAULTS_FILE = 47
exports.EE_FAILED_TO_HANDLE_DEFAULTS_FILE = 48
exports.EE_WRONG_DIRECTIVE_IN_CONFIG_FILE = 49
exports.EE_SKIPPING_DIRECTIVE_DUE_TO_MAX_INCLUDE_RECURSION = 50
exports.EE_INCORRECT_GRP_DEFINITION_IN_CONFIG_FILE = 51
exports.EE_OPTION_WITHOUT_GRP_IN_CONFIG_FILE = 52
exports.EE_CONFIG_FILE_PERMISSION_ERROR = 53
exports.EE_IGNORE_WORLD_WRITABLE_CONFIG_FILE = 54
exports.EE_USING_DISABLED_OPTION = 55
exports.EE_USING_DISABLED_SHORT_OPTION = 56
exports.EE_USING_PASSWORD_ON_CLI_IS_INSECURE = 57
exports.EE_UNKNOWN_SUFFIX_FOR_VARIABLE = 58
exports.EE_SSL_ERROR_FROM_FILE = 59
exports.EE_SSL_ERROR = 60
exports.EE_NET_SEND_ERROR_IN_BOOTSTRAP = 61
exports.EE_PACKETS_OUT_OF_ORDER = 62
exports.EE_UNKNOWN_PROTOCOL_OPTION = 63
exports.EE_FAILED_TO_LOCATE_SERVER_PUBLIC_KEY = 64
exports.EE_PUBLIC_KEY_NOT_IN_PEM_FORMAT = 65
exports.EE_DEBUG_INFO = 66
exports.EE_UNKNOWN_VARIABLE = 67
exports.EE_UNKNOWN_OPTION = 68
exports.EE_UNKNOWN_SHORT_OPTION = 69
exports.EE_OPTION_WITHOUT_ARGUMENT = 70
exports.EE_OPTION_REQUIRES_ARGUMENT = 71
exports.EE_SHORT_OPTION_REQUIRES_ARGUMENT = 72
exports.EE_OPTION_IGNORED_DUE_TO_INVALID_VALUE = 73
exports.EE_OPTION_WITH_EMPTY_VALUE = 74
exports.EE_FAILED_TO_ASSIGN_MAX_VALUE_TO_OPTION = 75
exports.EE_INCORRECT_BOOLEAN_VALUE_FOR_OPTION = 76
exports.EE_FAILED_TO_SET_OPTION_VALUE = 77
exports.EE_INCORRECT_INT_VALUE_FOR_OPTION = 78
exports.EE_INCORRECT_UINT_VALUE_FOR_OPTION = 79
exports.EE_ADJUSTED_SIGNED_VALUE_FOR_OPTION = 80
exports.EE_ADJUSTED_UNSIGNED_VALUE_FOR_OPTION = 81
exports.EE_ADJUSTED_ULONGLONG_VALUE_FOR_OPTION = 82
exports.EE_ADJUSTED_DOUBLE_VALUE_FOR_OPTION = 83
exports.EE_INVALID_DECIMAL_VALUE_FOR_OPTION = 84
exports.EE_COLLATION_PARSER_ERROR = 85
exports.EE_FAILED_TO_RESET_BEFORE_PRIMARY_IGNORABLE_CHAR = 86
exports.EE_FAILED_TO_RESET_BEFORE_TERTIARY_IGNORABLE_CHAR = 87
exports.EE_SHIFT_CHAR_OUT_OF_RANGE = 88
exports.EE_RESET_CHAR_OUT_OF_RANGE = 89
exports.EE_UNKNOWN_LDML_TAG = 90
exports.EE_FAILED_TO_RESET_BEFORE_SECONDARY_IGNORABLE_CHAR = 91
exports.EE_FAILED_PROCESSING_DIRECTIVE = 92
exports.EE_PTHREAD_KILL_FAILED = 93
exports.HA_ERR_KEY_NOT_FOUND = 120
exports.HA_ERR_FOUND_DUPP_KEY = 121
exports.HA_ERR_INTERNAL_ERROR = 122
exports.HA_ERR_RECORD_CHANGED = 123
exports.HA_ERR_WRONG_INDEX = 124
exports.HA_ERR_ROLLED_BACK = 125
exports.HA_ERR_CRASHED = 126
exports.HA_ERR_WRONG_IN_RECORD = 127
exports.HA_ERR_OUT_OF_MEM = 128
exports.HA_ERR_NOT_A_TABLE = 130
exports.HA_ERR_WRONG_COMMAND = 131
exports.HA_ERR_OLD_FILE = 132
exports.HA_ERR_NO_ACTIVE_RECORD = 133
exports.HA_ERR_RECORD_DELETED = 134
exports.HA_ERR_RECORD_FILE_FULL = 135
exports.HA_ERR_INDEX_FILE_FULL = 136
exports.HA_ERR_END_OF_FILE = 137
exports.HA_ERR_UNSUPPORTED = 138
exports.HA_ERR_TOO_BIG_ROW = 139
exports.HA_WRONG_CREATE_OPTION = 140
exports.HA_ERR_FOUND_DUPP_UNIQUE = 141
exports.HA_ERR_UNKNOWN_CHARSET = 142
exports.HA_ERR_WRONG_MRG_TABLE_DEF = 143
exports.HA_ERR_CRASHED_ON_REPAIR = 144
exports.HA_ERR_CRASHED_ON_USAGE = 145
exports.HA_ERR_LOCK_WAIT_TIMEOUT = 146
exports.HA_ERR_LOCK_TABLE_FULL = 147
exports.HA_ERR_READ_ONLY_TRANSACTION = 148
exports.HA_ERR_LOCK_DEADLOCK = 149
exports.HA_ERR_CANNOT_ADD_FOREIGN = 150
exports.HA_ERR_NO_REFERENCED_ROW = 151
exports.HA_ERR_ROW_IS_REFERENCED = 152
exports.HA_ERR_NO_SAVEPOINT = 153
exports.HA_ERR_NON_UNIQUE_BLOCK_SIZE = 154
exports.HA_ERR_NO_SUCH_TABLE = 155
exports.HA_ERR_TABLE_EXIST = 156
exports.HA_ERR_NO_CONNECTION = 157
exports.HA_ERR_NULL_IN_SPATIAL = 158
exports.HA_ERR_TABLE_DEF_CHANGED = 159
exports.HA_ERR_NO_PARTITION_FOUND = 160
exports.HA_ERR_RBR_LOGGING_FAILED = 161
exports.HA_ERR_DROP_INDEX_FK = 162
exports.HA_ERR_FOREIGN_DUPLICATE_KEY = 163
exports.HA_ERR_TABLE_NEEDS_UPGRADE = 164
exports.HA_ERR_TABLE_READONLY = 165
exports.HA_ERR_AUTOINC_READ_FAILED = 166
exports.HA_ERR_AUTOINC_ERANGE = 167
exports.HA_ERR_GENERIC = 168
exports.HA_ERR_RECORD_IS_THE_SAME = 169
exports.HA_ERR_LOGGING_IMPOSSIBLE = 170
exports.HA_ERR_CORRUPT_EVENT = 171
exports.HA_ERR_NEW_FILE = 172
exports.HA_ERR_ROWS_EVENT_APPLY = 173
exports.HA_ERR_INITIALIZATION = 174
exports.HA_ERR_FILE_TOO_SHORT = 175
exports.HA_ERR_WRONG_CRC = 176
exports.HA_ERR_TOO_MANY_CONCURRENT_TRXS = 177
exports.HA_ERR_NOT_IN_LOCK_PARTITIONS = 178
exports.HA_ERR_INDEX_COL_TOO_LONG = 179
exports.HA_ERR_INDEX_CORRUPT = 180
exports.HA_ERR_UNDO_REC_TOO_BIG = 181
exports.HA_FTS_INVALID_DOCID = 182
exports.HA_ERR_TABLE_IN_FK_CHECK = 183
exports.HA_ERR_TABLESPACE_EXISTS = 184
exports.HA_ERR_TOO_MANY_FIELDS = 185
exports.HA_ERR_ROW_IN_WRONG_PARTITION = 186
exports.HA_ERR_INNODB_READ_ONLY = 187
exports.HA_ERR_FTS_EXCEED_RESULT_CACHE_LIMIT = 188
exports.HA_ERR_TEMP_FILE_WRITE_FAILURE = 189
exports.HA_ERR_INNODB_FORCED_RECOVERY = 190
exports.HA_ERR_FTS_TOO_MANY_WORDS_IN_PHRASE = 191
exports.HA_ERR_FK_DEPTH_EXCEEDED = 192
exports.HA_MISSING_CREATE_OPTION = 193
exports.HA_ERR_SE_OUT_OF_MEMORY = 194
exports.HA_ERR_TABLE_CORRUPT = 195
exports.HA_ERR_QUERY_INTERRUPTED = 196
exports.HA_ERR_TABLESPACE_MISSING = 197
exports.HA_ERR_TABLESPACE_IS_NOT_EMPTY = 198
exports.HA_ERR_WRONG_FILE_NAME = 199
exports.HA_ERR_NOT_ALLOWED_COMMAND = 200
exports.HA_ERR_COMPUTE_FAILED = 201
exports.HA_ERR_ROW_FORMAT_CHANGED = 202
exports.HA_ERR_NO_WAIT_LOCK = 203
exports.HA_ERR_DISK_FULL_NOWAIT = 204
exports.HA_ERR_NO_SESSION_TEMP = 205
exports.HA_ERR_WRONG_TABLE_NAME = 206
exports.HA_ERR_TOO_LONG_PATH = 207
exports.HA_ERR_SAMPLING_INIT_FAILED = 208
exports.HA_ERR_FTS_TOO_MANY_NESTED_EXP = 209
exports.ER_HASHCHK = 1000
exports.ER_NISAMCHK = 1001
exports.ER_NO = 1002
exports.ER_YES = 1003
exports.ER_CANT_CREATE_FILE = 1004
exports.ER_CANT_CREATE_TABLE = 1005
exports.ER_CANT_CREATE_DB = 1006
exports.ER_DB_CREATE_EXISTS = 1007
exports.ER_DB_DROP_EXISTS = 1008
exports.ER_DB_DROP_DELETE = 1009
exports.ER_DB_DROP_RMDIR = 1010
exports.ER_CANT_DELETE_FILE = 1011
exports.ER_CANT_FIND_SYSTEM_REC = 1012
exports.ER_CANT_GET_STAT = 1013
exports.ER_CANT_GET_WD = 1014
exports.ER_CANT_LOCK = 1015
exports.ER_CANT_OPEN_FILE = 1016
exports.ER_FILE_NOT_FOUND = 1017
exports.ER_CANT_READ_DIR = 1018
exports.ER_CANT_SET_WD = 1019
exports.ER_CHECKREAD = 1020
exports.ER_DISK_FULL = 1021
exports.ER_DUP_KEY = 1022
exports.ER_ERROR_ON_CLOSE = 1023
exports.ER_ERROR_ON_READ = 1024
exports.ER_ERROR_ON_RENAME = 1025
exports.ER_ERROR_ON_WRITE = 1026
exports.ER_FILE_USED = 1027
exports.ER_FILSORT_ABORT = 1028
exports.ER_FORM_NOT_FOUND = 1029
exports.ER_GET_ERRNO = 1030
exports.ER_ILLEGAL_HA = 1031
exports.ER_KEY_NOT_FOUND = 1032
exports.ER_NOT_FORM_FILE = 1033
exports.ER_NOT_KEYFILE = 1034
exports.ER_OLD_KEYFILE = 1035
exports.ER_OPEN_AS_READONLY = 1036
exports.ER_OUTOFMEMORY = 1037
exports.ER_OUT_OF_SORTMEMORY = 1038
exports.ER_UNEXPECTED_EOF = 1039
exports.ER_CON_COUNT_ERROR = 1040
exports.ER_OUT_OF_RESOURCES = 1041
exports.ER_BAD_HOST_ERROR = 1042
exports.ER_HANDSHAKE_ERROR = 1043
exports.ER_DBACCESS_DENIED_ERROR = 1044
exports.ER_ACCESS_DENIED_ERROR = 1045
exports.ER_NO_DB_ERROR = 1046
exports.ER_UNKNOWN_COM_ERROR = 1047
exports.ER_BAD_NULL_ERROR = 1048
exports.ER_BAD_DB_ERROR = 1049
exports.ER_TABLE_EXISTS_ERROR = 1050
exports.ER_BAD_TABLE_ERROR = 1051
exports.ER_NON_UNIQ_ERROR = 1052
exports.ER_SERVER_SHUTDOWN = 1053
exports.ER_BAD_FIELD_ERROR = 1054
exports.ER_WRONG_FIELD_WITH_GROUP = 1055
exports.ER_WRONG_GROUP_FIELD = 1056
exports.ER_WRONG_SUM_SELECT = 1057
exports.ER_WRONG_VALUE_COUNT = 1058
exports.ER_TOO_LONG_IDENT = 1059
exports.ER_DUP_FIELDNAME = 1060
exports.ER_DUP_KEYNAME = 1061
exports.ER_DUP_ENTRY = 1062
exports.ER_WRONG_FIELD_SPEC = 1063
exports.ER_PARSE_ERROR = 1064
exports.ER_EMPTY_QUERY = 1065
exports.ER_NONUNIQ_TABLE = 1066
exports.ER_INVALID_DEFAULT = 1067
exports.ER_MULTIPLE_PRI_KEY = 1068
exports.ER_TOO_MANY_KEYS = 1069
exports.ER_TOO_MANY_KEY_PARTS = 1070
exports.ER_TOO_LONG_KEY = 1071
exports.ER_KEY_COLUMN_DOES_NOT_EXITS = 1072
exports.ER_BLOB_USED_AS_KEY = 1073
exports.ER_TOO_BIG_FIELDLENGTH = 1074
exports.ER_WRONG_AUTO_KEY = 1075
exports.ER_READY = 1076
exports.ER_NORMAL_SHUTDOWN = 1077
exports.ER_GOT_SIGNAL = 1078
exports.ER_SHUTDOWN_COMPLETE = 1079
exports.ER_FORCING_CLOSE = 1080
exports.ER_IPSOCK_ERROR = 1081
exports.ER_NO_SUCH_INDEX = 1082
exports.ER_WRONG_FIELD_TERMINATORS = 1083
exports.ER_BLOBS_AND_NO_TERMINATED = 1084
exports.ER_TEXTFILE_NOT_READABLE = 1085
exports.ER_FILE_EXISTS_ERROR = 1086
exports.ER_LOAD_INFO = 1087
exports.ER_ALTER_INFO = 1088
exports.ER_WRONG_SUB_KEY = 1089
exports.ER_CANT_REMOVE_ALL_FIELDS = 1090
exports.ER_CANT_DROP_FIELD_OR_KEY = 1091
exports.ER_INSERT_INFO = 1092
exports.ER_UPDATE_TABLE_USED = 1093
exports.ER_NO_SUCH_THREAD = 1094
exports.ER_KILL_DENIED_ERROR = 1095
exports.ER_NO_TABLES_USED = 1096
exports.ER_TOO_BIG_SET = 1097
exports.ER_NO_UNIQUE_LOGFILE = 1098
exports.ER_TABLE_NOT_LOCKED_FOR_WRITE = 1099
exports.ER_TABLE_NOT_LOCKED = 1100
exports.ER_BLOB_CANT_HAVE_DEFAULT = 1101
exports.ER_WRONG_DB_NAME = 1102
exports.ER_WRONG_TABLE_NAME = 1103
exports.ER_TOO_BIG_SELECT = 1104
exports.ER_UNKNOWN_ERROR = 1105
exports.ER_UNKNOWN_PROCEDURE = 1106
exports.ER_WRONG_PARAMCOUNT_TO_PROCEDURE = 1107
exports.ER_WRONG_PARAMETERS_TO_PROCEDURE = 1108
exports.ER_UNKNOWN_TABLE = 1109
exports.ER_FIELD_SPECIFIED_TWICE = 1110
exports.ER_INVALID_GROUP_FUNC_USE = 1111
exports.ER_UNSUPPORTED_EXTENSION = 1112
exports.ER_TABLE_MUST_HAVE_COLUMNS = 1113
exports.ER_RECORD_FILE_FULL = 1114
exports.ER_UNKNOWN_CHARACTER_SET = 1115
exports.ER_TOO_MANY_TABLES = 1116
exports.ER_TOO_MANY_FIELDS = 1117
exports.ER_TOO_BIG_ROWSIZE = 1118
exports.ER_STACK_OVERRUN = 1119
exports.ER_WRONG_OUTER_JOIN = 1120
exports.ER_NULL_COLUMN_IN_INDEX = 1121
exports.ER_CANT_FIND_UDF = 1122
exports.ER_CANT_INITIALIZE_UDF = 1123
exports.ER_UDF_NO_PATHS = 1124
exports.ER_UDF_EXISTS = 1125
exports.ER_CANT_OPEN_LIBRARY = 1126
exports.ER_CANT_FIND_DL_ENTRY = 1127
exports.ER_FUNCTION_NOT_DEFINED = 1128
exports.ER_HOST_IS_BLOCKED = 1129
exports.ER_HOST_NOT_PRIVILEGED = 1130
exports.ER_PASSWORD_ANONYMOUS_USER = 1131
exports.ER_PASSWORD_NOT_ALLOWED = 1132
exports.ER_PASSWORD_NO_MATCH = 1133
exports.ER_UPDATE_INFO = 1134
exports.ER_CANT_CREATE_THREAD = 1135
exports.ER_WRONG_VALUE_COUNT_ON_ROW = 1136
exports.ER_CANT_REOPEN_TABLE = 1137
exports.ER_INVALID_USE_OF_NULL = 1138
exports.ER_REGEXP_ERROR = 1139
exports.ER_MIX_OF_GROUP_FUNC_AND_FIELDS = 1140
exports.ER_NONEXISTING_GRANT = 1141
exports.ER_TABLEACCESS_DENIED_ERROR = 1142
exports.ER_COLUMNACCESS_DENIED_ERROR = 1143
exports.ER_ILLEGAL_GRANT_FOR_TABLE = 1144
exports.ER_GRANT_WRONG_HOST_OR_USER = 1145
exports.ER_NO_SUCH_TABLE = 1146
exports.ER_NONEXISTING_TABLE_GRANT = 1147
exports.ER_NOT_ALLOWED_COMMAND = 1148
exports.ER_SYNTAX_ERROR = 1149
exports.ER_UNUSED1 = 1150
exports.ER_UNUSED2 = 1151
exports.ER_ABORTING_CONNECTION = 1152
exports.ER_NET_PACKET_TOO_LARGE = 1153
exports.ER_NET_READ_ERROR_FROM_PIPE = 1154
exports.ER_NET_FCNTL_ERROR = 1155
exports.ER_NET_PACKETS_OUT_OF_ORDER = 1156
exports.ER_NET_UNCOMPRESS_ERROR = 1157
exports.ER_NET_READ_ERROR = 1158
exports.ER_NET_READ_INTERRUPTED = 1159
exports.ER_NET_ERROR_ON_WRITE = 1160
exports.ER_NET_WRITE_INTERRUPTED = 1161
exports.ER_TOO_LONG_STRING = 1162
exports.ER_TABLE_CANT_HANDLE_BLOB = 1163
exports.ER_TABLE_CANT_HANDLE_AUTO_INCREMENT = 1164
exports.ER_UNUSED3 = 1165
exports.ER_WRONG_COLUMN_NAME = 1166
exports.ER_WRONG_KEY_COLUMN = 1167
exports.ER_WRONG_MRG_TABLE = 1168
exports.ER_DUP_UNIQUE = 1169
exports.ER_BLOB_KEY_WITHOUT_LENGTH = 1170
exports.ER_PRIMARY_CANT_HAVE_NULL = 1171
exports.ER_TOO_MANY_ROWS = 1172
exports.ER_REQUIRES_PRIMARY_KEY = 1173
exports.ER_NO_RAID_COMPILED = 1174
exports.ER_UPDATE_WITHOUT_KEY_IN_SAFE_MODE = 1175
exports.ER_KEY_DOES_NOT_EXITS = 1176
exports.ER_CHECK_NO_SUCH_TABLE = 1177
exports.ER_CHECK_NOT_IMPLEMENTED = 1178
exports.ER_CANT_DO_THIS_DURING_AN_TRANSACTION = 1179
exports.ER_ERROR_DURING_COMMIT = 1180
exports.ER_ERROR_DURING_ROLLBACK = 1181
exports.ER_ERROR_DURING_FLUSH_LOGS = 1182
exports.ER_ERROR_DURING_CHECKPOINT = 1183
exports.ER_NEW_ABORTING_CONNECTION = 1184
exports.ER_DUMP_NOT_IMPLEMENTED = 1185
exports.ER_FLUSH_MASTER_BINLOG_CLOSED = 1186
exports.ER_INDEX_REBUILD = 1187
exports.ER_SOURCE = 1188
exports.ER_SOURCE_NET_READ = 1189
exports.ER_SOURCE_NET_WRITE = 1190
exports.ER_FT_MATCHING_KEY_NOT_FOUND = 1191
exports.ER_LOCK_OR_ACTIVE_TRANSACTION = 1192
exports.ER_UNKNOWN_SYSTEM_VARIABLE = 1193
exports.ER_CRASHED_ON_USAGE = 1194
exports.ER_CRASHED_ON_REPAIR = 1195
exports.ER_WARNING_NOT_COMPLETE_ROLLBACK = 1196
exports.ER_TRANS_CACHE_FULL = 1197
exports.ER_SLAVE_MUST_STOP = 1198
exports.ER_REPLICA_NOT_RUNNING = 1199
exports.ER_BAD_REPLICA = 1200
exports.ER_CONNECTION_METADATA = 1201
exports.ER_REPLICA_THREAD = 1202
exports.ER_TOO_MANY_USER_CONNECTIONS = 1203
exports.ER_SET_CONSTANTS_ONLY = 1204
exports.ER_LOCK_WAIT_TIMEOUT = 1205
exports.ER_LOCK_TABLE_FULL = 1206
exports.ER_READ_ONLY_TRANSACTION = 1207
exports.ER_DROP_DB_WITH_READ_LOCK = 1208
exports.ER_CREATE_DB_WITH_READ_LOCK = 1209
exports.ER_WRONG_ARGUMENTS = 1210
exports.ER_NO_PERMISSION_TO_CREATE_USER = 1211
exports.ER_UNION_TABLES_IN_DIFFERENT_DIR = 1212
exports.ER_LOCK_DEADLOCK = 1213
exports.ER_TABLE_CANT_HANDLE_FT = 1214
exports.ER_CANNOT_ADD_FOREIGN = 1215
exports.ER_NO_REFERENCED_ROW = 1216
exports.ER_ROW_IS_REFERENCED = 1217
exports.ER_CONNECT_TO_SOURCE = 1218
exports.ER_QUERY_ON_MASTER = 1219
exports.ER_ERROR_WHEN_EXECUTING_COMMAND = 1220
exports.ER_WRONG_USAGE = 1221
exports.ER_WRONG_NUMBER_OF_COLUMNS_IN_SELECT = 1222
exports.ER_CANT_UPDATE_WITH_READLOCK = 1223
exports.ER_MIXING_NOT_ALLOWED = 1224
exports.ER_DUP_ARGUMENT = 1225
exports.ER_USER_LIMIT_REACHED = 1226
exports.ER_SPECIFIC_ACCESS_DENIED_ERROR = 1227
exports.ER_LOCAL_VARIABLE = 1228
exports.ER_GLOBAL_VARIABLE = 1229
exports.ER_NO_DEFAULT = 1230
exports.ER_WRONG_VALUE_FOR_VAR = 1231
exports.ER_WRONG_TYPE_FOR_VAR = 1232
exports.ER_VAR_CANT_BE_READ = 1233
exports.ER_CANT_USE_OPTION_HERE = 1234
exports.ER_NOT_SUPPORTED_YET = 1235
exports.ER_SOURCE_FATAL_ERROR_READING_BINLOG = 1236
exports.ER_REPLICA_IGNORED_TABLE = 1237
exports.ER_INCORRECT_GLOBAL_LOCAL_VAR = 1238
exports.ER_WRONG_FK_DEF = 1239
exports.ER_KEY_REF_DO_NOT_MATCH_TABLE_REF = 1240
exports.ER_OPERAND_COLUMNS = 1241
exports.ER_SUBQUERY_NO_1_ROW = 1242
exports.ER_UNKNOWN_STMT_HANDLER = 1243
exports.ER_CORRUPT_HELP_DB = 1244
exports.ER_CYCLIC_REFERENCE = 1245
exports.ER_AUTO_CONVERT = 1246
exports.ER_ILLEGAL_REFERENCE = 1247
exports.ER_DERIVED_MUST_HAVE_ALIAS = 1248
exports.ER_SELECT_REDUCED = 1249
exports.ER_TABLENAME_NOT_ALLOWED_HERE = 1250
exports.ER_NOT_SUPPORTED_AUTH_MODE = 1251
exports.ER_SPATIAL_CANT_HAVE_NULL = 1252
exports.ER_COLLATION_CHARSET_MISMATCH = 1253
exports.ER_SLAVE_WAS_RUNNING = 1254
exports.ER_SLAVE_WAS_NOT_RUNNING = 1255
exports.ER_TOO_BIG_FOR_UNCOMPRESS = 1256
exports.ER_ZLIB_Z_MEM_ERROR = 1257
exports.ER_ZLIB_Z_BUF_ERROR = 1258
exports.ER_ZLIB_Z_DATA_ERROR = 1259
exports.ER_CUT_VALUE_GROUP_CONCAT = 1260
exports.ER_WARN_TOO_FEW_RECORDS = 1261
exports.ER_WARN_TOO_MANY_RECORDS = 1262
exports.ER_WARN_NULL_TO_NOTNULL = 1263
exports.ER_WARN_DATA_OUT_OF_RANGE = 1264
exports.WARN_DATA_TRUNCATED = 1265
exports.ER_WARN_USING_OTHER_HANDLER = 1266
exports.ER_CANT_AGGREGATE_2COLLATIONS = 1267
exports.ER_DROP_USER = 1268
exports.ER_REVOKE_GRANTS = 1269
exports.ER_CANT_AGGREGATE_3COLLATIONS = 1270
exports.ER_CANT_AGGREGATE_NCOLLATIONS = 1271
exports.ER_VARIABLE_IS_NOT_STRUCT = 1272
exports.ER_UNKNOWN_COLLATION = 1273
exports.ER_REPLICA_IGNORED_SSL_PARAMS = 1274
exports.ER_SERVER_IS_IN_SECURE_AUTH_MODE = 1275
exports.ER_WARN_FIELD_RESOLVED = 1276
exports.ER_BAD_REPLICA_UNTIL_COND = 1277
exports.ER_MISSING_SKIP_REPLICA = 1278
exports.ER_UNTIL_COND_IGNORED = 1279
exports.ER_WRONG_NAME_FOR_INDEX = 1280
exports.ER_WRONG_NAME_FOR_CATALOG = 1281
exports.ER_WARN_QC_RESIZE = 1282
exports.ER_BAD_FT_COLUMN = 1283
exports.ER_UNKNOWN_KEY_CACHE = 1284
exports.ER_WARN_HOSTNAME_WONT_WORK = 1285
exports.ER_UNKNOWN_STORAGE_ENGINE = 1286
exports.ER_WARN_DEPRECATED_SYNTAX = 1287
exports.ER_NON_UPDATABLE_TABLE = 1288
exports.ER_FEATURE_DISABLED = 1289
exports.ER_OPTION_PREVENTS_STATEMENT = 1290
exports.ER_DUPLICATED_VALUE_IN_TYPE = 1291
exports.ER_TRUNCATED_WRONG_VALUE = 1292
exports.ER_TOO_MUCH_AUTO_TIMESTAMP_COLS = 1293
exports.ER_INVALID_ON_UPDATE = 1294
exports.ER_UNSUPPORTED_PS = 1295
exports.ER_GET_ERRMSG = 1296
exports.ER_GET_TEMPORARY_ERRMSG = 1297
exports.ER_UNKNOWN_TIME_ZONE = 1298
exports.ER_WARN_INVALID_TIMESTAMP = 1299
exports.ER_INVALID_CHARACTER_STRING = 1300
exports.ER_WARN_ALLOWED_PACKET_OVERFLOWED = 1301
exports.ER_CONFLICTING_DECLARATIONS = 1302
exports.ER_SP_NO_RECURSIVE_CREATE = 1303
exports.ER_SP_ALREADY_EXISTS = 1304
exports.ER_SP_DOES_NOT_EXIST = 1305
exports.ER_SP_DROP_FAILED = 1306
exports.ER_SP_STORE_FAILED = 1307
exports.ER_SP_LILABEL_MISMATCH = 1308
exports.ER_SP_LABEL_REDEFINE = 1309
exports.ER_SP_LABEL_MISMATCH = 1310
exports.ER_SP_UNINIT_VAR = 1311
exports.ER_SP_BADSELECT = 1312
exports.ER_SP_BADRETURN = 1313
exports.ER_SP_BADSTATEMENT = 1314
exports.ER_UPDATE_LOG_DEPRECATED_IGNORED = 1315
exports.ER_UPDATE_LOG_DEPRECATED_TRANSLATED = 1316
exports.ER_QUERY_INTERRUPTED = 1317
exports.ER_SP_WRONG_NO_OF_ARGS = 1318
exports.ER_SP_COND_MISMATCH = 1319
exports.ER_SP_NORETURN = 1320
exports.ER_SP_NORETURNEND = 1321
exports.ER_SP_BAD_CURSOR_QUERY = 1322
exports.ER_SP_BAD_CURSOR_SELECT = 1323
exports.ER_SP_CURSOR_MISMATCH = 1324
exports.ER_SP_CURSOR_ALREADY_OPEN = 1325
exports.ER_SP_CURSOR_NOT_OPEN = 1326
exports.ER_SP_UNDECLARED_VAR = 1327
exports.ER_SP_WRONG_NO_OF_FETCH_ARGS = 1328
exports.ER_SP_FETCH_NO_DATA = 1329
exports.ER_SP_DUP_PARAM = 1330
exports.ER_SP_DUP_VAR = 1331
exports.ER_SP_DUP_COND = 1332
exports.ER_SP_DUP_CURS = 1333
exports.ER_SP_CANT_ALTER = 1334
exports.ER_SP_SUBSELECT_NYI = 1335
exports.ER_STMT_NOT_ALLOWED_IN_SF_OR_TRG = 1336
exports.ER_SP_VARCOND_AFTER_CURSHNDLR = 1337
exports.ER_SP_CURSOR_AFTER_HANDLER = 1338
exports.ER_SP_CASE_NOT_FOUND = 1339
exports.ER_FPARSER_TOO_BIG_FILE = 1340
exports.ER_FPARSER_BAD_HEADER = 1341
exports.ER_FPARSER_EOF_IN_COMMENT = 1342
exports.ER_FPARSER_ERROR_IN_PARAMETER = 1343
exports.ER_FPARSER_EOF_IN_UNKNOWN_PARAMETER = 1344
exports.ER_VIEW_NO_EXPLAIN = 1345
exports.ER_FRM_UNKNOWN_TYPE = 1346
exports.ER_WRONG_OBJECT = 1347
exports.ER_NONUPDATEABLE_COLUMN = 1348
exports.ER_VIEW_SELECT_DERIVED = 1349
exports.ER_VIEW_SELECT_CLAUSE = 1350
exports.ER_VIEW_SELECT_VARIABLE = 1351
exports.ER_VIEW_SELECT_TMPTABLE = 1352
exports.ER_VIEW_WRONG_LIST = 1353
exports.ER_WARN_VIEW_MERGE = 1354
exports.ER_WARN_VIEW_WITHOUT_KEY = 1355
exports.ER_VIEW_INVALID = 1356
exports.ER_SP_NO_DROP_SP = 1357
exports.ER_SP_GOTO_IN_HNDLR = 1358
exports.ER_TRG_ALREADY_EXISTS = 1359
exports.ER_TRG_DOES_NOT_EXIST = 1360
exports.ER_TRG_ON_VIEW_OR_TEMP_TABLE = 1361
exports.ER_TRG_CANT_CHANGE_ROW = 1362
exports.ER_TRG_NO_SUCH_ROW_IN_TRG = 1363
exports.ER_NO_DEFAULT_FOR_FIELD = 1364
exports.ER_DIVISION_BY_ZERO = 1365
exports.ER_TRUNCATED_WRONG_VALUE_FOR_FIELD = 1366
exports.ER_ILLEGAL_VALUE_FOR_TYPE = 1367
exports.ER_VIEW_NONUPD_CHECK = 1368
exports.ER_VIEW_CHECK_FAILED = 1369
exports.ER_PROCACCESS_DENIED_ERROR = 1370
exports.ER_RELAY_LOG_FAIL = 1371
exports.ER_PASSWD_LENGTH = 1372
exports.ER_UNKNOWN_TARGET_BINLOG = 1373
exports.ER_IO_ERR_LOG_INDEX_READ = 1374
exports.ER_BINLOG_PURGE_PROHIBITED = 1375
exports.ER_FSEEK_FAIL = 1376
exports.ER_BINLOG_PURGE_FATAL_ERR = 1377
exports.ER_LOG_IN_USE = 1378
exports.ER_LOG_PURGE_UNKNOWN_ERR = 1379
exports.ER_RELAY_LOG_INIT = 1380
exports.ER_NO_BINARY_LOGGING = 1381
exports.ER_RESERVED_SYNTAX = 1382
exports.ER_WSAS_FAILED = 1383
exports.ER_DIFF_GROUPS_PROC = 1384
exports.ER_NO_GROUP_FOR_PROC = 1385
exports.ER_ORDER_WITH_PROC = 1386
exports.ER_LOGGING_PROHIBIT_CHANGING_OF = 1387
exports.ER_NO_FILE_MAPPING = 1388
exports.ER_WRONG_MAGIC = 1389
exports.ER_PS_MANY_PARAM = 1390
exports.ER_KEY_PART_0 = 1391
exports.ER_VIEW_CHECKSUM = 1392
exports.ER_VIEW_MULTIUPDATE = 1393
exports.ER_VIEW_NO_INSERT_FIELD_LIST = 1394
exports.ER_VIEW_DELETE_MERGE_VIEW = 1395
exports.ER_CANNOT_USER = 1396
exports.ER_XAER_NOTA = 1397
exports.ER_XAER_INVAL = 1398
exports.ER_XAER_RMFAIL = 1399
exports.ER_XAER_OUTSIDE = 1400
exports.ER_XAER_RMERR = 1401
exports.ER_XA_RBROLLBACK = 1402
exports.ER_NONEXISTING_PROC_GRANT = 1403
exports.ER_PROC_AUTO_GRANT_FAIL = 1404
exports.ER_PROC_AUTO_REVOKE_FAIL = 1405
exports.ER_DATA_TOO_LONG = 1406
exports.ER_SP_BAD_SQLSTATE = 1407
exports.ER_STARTUP = 1408
exports.ER_LOAD_FROM_FIXED_SIZE_ROWS_TO_VAR = 1409
exports.ER_CANT_CREATE_USER_WITH_GRANT = 1410
exports.ER_WRONG_VALUE_FOR_TYPE = 1411
exports.ER_TABLE_DEF_CHANGED = 1412
exports.ER_SP_DUP_HANDLER = 1413
exports.ER_SP_NOT_VAR_ARG = 1414
exports.ER_SP_NO_RETSET = 1415
exports.ER_CANT_CREATE_GEOMETRY_OBJECT = 1416
exports.ER_FAILED_ROUTINE_BREAK_BINLOG = 1417
exports.ER_BINLOG_UNSAFE_ROUTINE = 1418
exports.ER_BINLOG_CREATE_ROUTINE_NEED_SUPER = 1419
exports.ER_EXEC_STMT_WITH_OPEN_CURSOR = 1420
exports.ER_STMT_HAS_NO_OPEN_CURSOR = 1421
exports.ER_COMMIT_NOT_ALLOWED_IN_SF_OR_TRG = 1422
exports.ER_NO_DEFAULT_FOR_VIEW_FIELD = 1423
exports.ER_SP_NO_RECURSION = 1424
exports.ER_TOO_BIG_SCALE = 1425
exports.ER_TOO_BIG_PRECISION = 1426
exports.ER_M_BIGGER_THAN_D = 1427
exports.ER_WRONG_LOCK_OF_SYSTEM_TABLE = 1428
exports.ER_CONNECT_TO_FOREIGN_DATA_SOURCE = 1429
exports.ER_QUERY_ON_FOREIGN_DATA_SOURCE = 1430
exports.ER_FOREIGN_DATA_SOURCE_DOESNT_EXIST = 1431
exports.ER_FOREIGN_DATA_STRING_INVALID_CANT_CREATE = 1432
exports.ER_FOREIGN_DATA_STRING_INVALID = 1433
exports.ER_CANT_CREATE_FEDERATED_TABLE = 1434
exports.ER_TRG_IN_WRONG_SCHEMA = 1435
exports.ER_STACK_OVERRUN_NEED_MORE = 1436
exports.ER_TOO_LONG_BODY = 1437
exports.ER_WARN_CANT_DROP_DEFAULT_KEYCACHE = 1438
exports.ER_TOO_BIG_DISPLAYWIDTH = 1439
exports.ER_XAER_DUPID = 1440
exports.ER_DATETIME_FUNCTION_OVERFLOW = 1441
exports.ER_CANT_UPDATE_USED_TABLE_IN_SF_OR_TRG = 1442
exports.ER_VIEW_PREVENT_UPDATE = 1443
exports.ER_PS_NO_RECURSION = 1444
exports.ER_SP_CANT_SET_AUTOCOMMIT = 1445
exports.ER_MALFORMED_DEFINER = 1446
exports.ER_VIEW_FRM_NO_USER = 1447
exports.ER_VIEW_OTHER_USER = 1448
exports.ER_NO_SUCH_USER = 1449
exports.ER_FORBID_SCHEMA_CHANGE = 1450
exports.ER_ROW_IS_REFERENCED_2 = 1451
exports.ER_NO_REFERENCED_ROW_2 = 1452
exports.ER_SP_BAD_VAR_SHADOW = 1453
exports.ER_TRG_NO_DEFINER = 1454
exports.ER_OLD_FILE_FORMAT = 1455
exports.ER_SP_RECURSION_LIMIT = 1456
exports.ER_SP_PROC_TABLE_CORRUPT = 1457
exports.ER_SP_WRONG_NAME = 1458
exports.ER_TABLE_NEEDS_UPGRADE = 1459
exports.ER_SP_NO_AGGREGATE = 1460
exports.ER_MAX_PREPARED_STMT_COUNT_REACHED = 1461
exports.ER_VIEW_RECURSIVE = 1462
exports.ER_NON_GROUPING_FIELD_USED = 1463
exports.ER_TABLE_CANT_HANDLE_SPKEYS = 1464
exports.ER_NO_TRIGGERS_ON_SYSTEM_SCHEMA = 1465
exports.ER_REMOVED_SPACES = 1466
exports.ER_AUTOINC_READ_FAILED = 1467
exports.ER_USERNAME = 1468
exports.ER_HOSTNAME = 1469
exports.ER_WRONG_STRING_LENGTH = 1470
exports.ER_NON_INSERTABLE_TABLE = 1471
exports.ER_ADMIN_WRONG_MRG_TABLE = 1472
exports.ER_TOO_HIGH_LEVEL_OF_NESTING_FOR_SELECT = 1473
exports.ER_NAME_BECOMES_EMPTY = 1474
exports.ER_AMBIGUOUS_FIELD_TERM = 1475
exports.ER_FOREIGN_SERVER_EXISTS = 1476
exports.ER_FOREIGN_SERVER_DOESNT_EXIST = 1477
exports.ER_ILLEGAL_HA_CREATE_OPTION = 1478
exports.ER_PARTITION_REQUIRES_VALUES_ERROR = 1479
exports.ER_PARTITION_WRONG_VALUES_ERROR = 1480
exports.ER_PARTITION_MAXVALUE_ERROR = 1481
exports.ER_PARTITION_SUBPARTITION_ERROR = 1482
exports.ER_PARTITION_SUBPART_MIX_ERROR = 1483
exports.ER_PARTITION_WRONG_NO_PART_ERROR = 1484
exports.ER_PARTITION_WRONG_NO_SUBPART_ERROR = 1485
exports.ER_WRONG_EXPR_IN_PARTITION_FUNC_ERROR = 1486
exports.ER_NO_CONST_EXPR_IN_RANGE_OR_LIST_ERROR = 1487
